gpflow/base.py

```diff
@@ -159,8 +159,9 @@ def validate_unconstrained_value(self, value: tf.Tensor, dtype: DType) -> tf.Ten
         value = _cast_to_dtype(value, dtype)
         unconstrained_value = _to_unconstrained(value, self.transform)
         message = (
-            "gpflow.Parameter: unconstrained value of passed value "
-            "has NaN or Inf and cannot be assigned."
+            "gpflow.Parameter: the value to be assigned is incompatible with this parameter's "
+            "transform (the corresponding unconstrained value has NaN or Inf) and hence cannot be "
+            "assigned."
```

Comment:
Could be extra explicit and add `(e.g. positively constrained parameters must be strictly greater than zero)`? :thinking:

Reply:
`positive` is too specific for this message.

Reply:
it's the most common use-case for transforms, and examples can be helpful no?